manager/allocator/networkallocator/portallocator.go

```diff
@@ -91,13 +91,17 @@ func reconcilePortConfigs(s *api.Service) []*api.PortConfig {
 		return s.Spec.Endpoint.Ports
 	}
 
-	allocatedPorts := make(map[api.PortConfig]*api.PortConfig)
+	allocatedPorts := make(map[api.PortConfig]map[uint32]*api.PortConfig)
 	for _, portState := range s.Endpoint.Ports {
 		if portState.PublishMode != api.PublishModeIngress {
 			continue
 		}
 
-		allocatedPorts[getPortConfigKey(portState)] = portState
+		portKey := getPortConfigKey(portState)
+		if _, ok := allocatedPorts[portKey]; !ok {
+			allocatedPorts[portKey] = make(map[uint32]*api.PortConfig)
```

Comment:
I agree with @thaJeztah, even before considering the `-p X/80 -p Y/80` problem, already the `-p 80 -p 80` cannot be satisfied by the current logic, if I understand it correctly, because it relays on the assumption of a unique config per target port/protocol. This seem the root cause for both the issues (again IIUIC).
Maybe we should rethink the whole logic with that in mind from the beginning.
@yongtang does your fix also address the `-p 80 -p 80` use case ?